src/main.rs

```diff
@@ -303,3 +329,87 @@ async fn sync_tasks_to_everhour(
     }
     println!("Everhour user id: {}", me.id);
 }
+
+async fn sync_tasks_to_shtab(
+    hamster_db: Option<String>,
+    api_token: String,
+    activity_id: Option<i64>,
+    from: NaiveDate,
+    to: NaiveDate,
+    category: Option<String>,
+    run_mode: RunMode,
+) {
+    let client = ShtabClient::new(Some(api_token));
+    let me = client.get_profile().await.unwrap();
+
+    let activity_id:i64 = match activity_id {
+        Some(activity_id) => activity_id,
+        None => match std::env::var("HAMCLI_SHTAB_ACTIVITY") {
+            Ok(activity_id_str) => activity_id_str.parse::<i64>().unwrap(),
+            Err(_) => panic!(
+                "Please specify activity id via --activity-id option or HAMCLI_SHTAB_ACTIVITY env var")
+        }
+    };
+
+    let local_tz = Local::now().timezone();
+
+    let mut day = from;
+    while day <= to {
+        println!("Processing day {}", day);
+        let next_day = day.checked_add_days(Days::new(1)).unwrap();
+        let tasks = get_tasks_with_durations(hamster_db.clone(), day, next_day, category.clone());
+        let mut total_duration = Duration::new(0, 0);
+        let mut last_time = day
+            .and_hms_opt(0, 0, 0)
+            .unwrap()
+            .and_local_timezone(local_tz)
+            .unwrap();
+        for (task_id, task_data) in tasks.into_iter() {
+            let task_id: i64 = match task_id {
+                Some(task_id) => task_id.parse::<i64>().unwrap(),
+                None => match run_mode {
+                    RunMode::DryRun => -1,
+                    RunMode::Normal => panic!(
+                        "Missing task id! ({}, '{}')",
+                        task_data.duration.as_hhmm(),
+                        task_data.title.unwrap_or("-".to_string())
+                    ),
+                },
+            };
```

Comment:
```suggestion
            let Some(task_id: i64) = task_id.map(|id|id.parse::<i64>().unwrap()).or( (run_mode==RunMode::DryRun).then_some(-1)) else {
                    panic!(
                        "Missing task id! ({}, '{}')",
                        task_data.duration.as_hhmm(),
                        task_data.title.unwrap_or("-".to_string())
                    ),
            };
```
покороче, но не 100% понятней, конечно